### ISSUE TITLE
Wrong LightDM Font Scaling

### ISSUE THREAD

**Issue by wallace11** [stockmind/gpd-pocket-ubuntu-respin#89]
I've been around this getting Linux to work on the GPD Pocket thing pretty much from the start, and I remember that at some point (probably with Ansible) we had a LightDM screen on boot showing @ 1920x1080 with the correct hidpi font scaling, in contrary to the situation right now where it's showing at the native resolution with font scaling set to 1 (probably).
I've tried a couple of well-known tricks, from editing a `/com/canonical/unity-greeter/xft-dpi` value, to setting `xrandr` scripts to run on LDM load, to editing `monitors.xml`. None of them seem to work.
I've been putting this off for quite some time now, hoping that a fix will be found, but this is definitely a bug that should be addressed. I'm not really sure how to fix it though. 
On a side note, if I'm logged in and go back go the lock screen (`Win+L`) LDM shows with the correct font scaling (probably inheriting the setting from the running session). It indicates that a viable solution does exist.

**Comment by stockmind** [comment]
Could you post a screenshot or photo of the issue?

Also could you post output of problem-reporting script ?

Thank you

Il 08 gen 2018 22:24, "wallace11" <notifications@github.com> ha scritto:

**Comment by wallace11** [comment]
Yes. Extremely sorry for the lousy bug report.
![When booting up](https://i.imgur.com/YgFVbMx.jpg)
![Locking the screen after boot up](https://i.imgur.com/M5M8V8H.jpg)

And the output of `problem-reporting`:
```
**Kernel:** 4.15.0-rc5-custom-matlala-3-1-2018
**Desktop Environment:** 
**Display:** :0
**Monitor:** DSI1 connected primary 1920x1200+0+0 right (normal left inverted right x axis y axis) 95mm x 151mm
**Scripts:**  gpdfand gpdscreen gpdscreen-indicator
**EFI:**  [ 0.000000] efi: EFI v2.40 by American Megatrends
**EFI**  Release Date: 06/28/2017
```

**Comment by stockmind** [comment]
Odd, i have always right lightDM scaling at boot (HiDPI). I've seen problems like this when grub or intel xorg driver was not configured correctly. Could you post output of

    cat /etc/default/grub
    cat /etc/X11/xorg.conf.d/*.conf
    ls /etc/X11/xorg.conf.d/

Thanks

**Comment by wallace11** [comment]
Hi,
I'll post termbin links since it's gonna be long:
`cat /etc/default/grub.ucf-dist` (mind the extension): http://termbin.com/0pr7
`for f in /etc/X11/xorg.conf.d/*; do echo $f; cat $f; done` (a slight change, better result :P):  http://termbin.com/o16m

I'll also try to investigate further about this Intel driver issue...

**Comment by wallace11** [comment]
Did this information help?

**Comment by ferlanero** [comment]
Same here. Wrong LightDM font scalling on Ubuntu Budgie 17.10 Here my ISO respin wich is wrong: [https://drive.google.com/file/d/1NDJTATwCT2XByr2QRLW_nQn9jcQK07RD/view](https://drive.google.com/file/d/1NDJTATwCT2XByr2QRLW_nQn9jcQK07RD/view)

**Comment by stockmind** [comment]
@wallace11 Your system files seems fine, can you check this also?

    for f in /etc/X11/Xsession.d/*; do echo $f; cat $f; done

When was the last time you did a fresh install?
Your system come from an ansible configuration right?
There must be something wrong somewhere related to driver loading.. I can reproduce your issue if i add modesetting to grub kernel boot options or if i don't add the "intel.conf" (so modesetting driver will be used), but on intel driver the login screen is always on right DPI! 
Do you have some free space on which you can try a fresh install of ubuntu 17.04 unity just to be sure? Also are you on 17.04 or 16.04?
What desktop environment do you use as default?

@ferlanero Previous respin of Budgie were ok? (17.04)

**Comment by ferlanero** [comment]
@stockmind No. In my Ubuntu Budgie respins never worked the font scalling on LightDM :( And never figured why.

**Comment by wallace11** [comment]
@stockmind 
1. I created a TOC to make it easier: http://termbin.com/kbgo
2. Yeah I'm on the same installation as from before even the ansible project existed :( put a lot of work into making this a productive device, but I guess that at some point a fresh installation is going to be inevitable... Will probably have the time for that only next month though.
3. I purely hate Gnome, and making XFCE play nice in touch screen is a lot of work, so Unity it my choice for this little fellow ;)
4. I also checked everything I can to make sure the Intel driver runs on boot... No luck so far. There must be another unknown place that plays part in this...
